public/app/features/dashboard/components/ShareModal/SharePublicDashboardUtils.ts

```diff
@@ -1,21 +1,37 @@
 import { getBackendSrv } from '@grafana/runtime';
 import { DashboardModel } from 'app/features/dashboard/state';
+import { DashboardDataDTO, DashboardMeta } from 'app/types/dashboard';
 
 export interface PublicDashboardConfig {
   isPublic: boolean;
+  publicDashboard: {
+    uid: string;
+    dashboardUid: string;
+    orgId: number;
+    timeVariables?: object;
```

Comment:
timeVariables is not a good name, this is more like timeSettings. This will contain properties like canChange, limits on from and to (and default from and to). But this is not variables. 

Variables are something different.

Reply:
but I would rename this to just settings, so we can have one json property for time settings, and variable values (Variables are something different from the time range settings)